gui/layers/_markers_layer.py

```diff
@@ -260,16 +269,37 @@ def _slice_markers(self, indices):
         # Get a list of the coords for the markers in this slice
         coords = self.coords
         if len(coords) > 0:
-            matches = np.equal(
-                coords[:, 2:],
-                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
-
-            matches = np.all(matches, axis=1)
-
-            in_slice_markers = coords[matches, :2]
-            return in_slice_markers, matches
+            if self._n_dimensional is False:
+                matches = np.equal(
+                    coords[:, 2:],
+                    np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
+
+                matches = np.all(matches, axis=1)
+
+                in_slice_markers = coords[matches, :2]
+                return in_slice_markers, matches, 1
+            elif self.ndim > 2:
+                distances = abs(coords[:, 2:] -
+                                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
+                if isinstance(self.size, np.ndarray):
+                    if len(self.size) == len(indices):
+                        size_array = np.broadcast_to(self.size[2:]/2, (len(coords), len(indices) - 2))
+                    elif isinstance(self.size, np.ndarray) and len(self.size.shape)==2:
+                        size_array = self.size[:,2:]/2
+                    else:
+                        size_array = np.broadcast_to(self.size/2, (len(indices) - 2, len(coords))).T
+                else:
+                    size_array = np.broadcast_to(self.size/2, (len(indices) - 2, len(coords))).T
+
+                matches = np.less_equal(distances, size_array)
+                matches = np.all(matches, axis=1)
+
+                in_slice_markers = coords[matches, :2]
+                scale = (size_array[matches] - distances[matches])/size_array[matches]
+                scale = np.prod(scale, axis=1)
```

Comment:
Again, let's not reuse names. Suggestion:

```
scale_per_dim = (size_array[matches] - distances[matches]) / size_array[matches]
scale = np.prod(scale_per_dim, axis=1)
```

Reply:
fixed